Plans in Site Creation: Show full free domain explanation label in domain selection view

staskus opened the pull request:
Addressing the issue of not showing the full "Free for the first year with annual paid plans" (https://github.com/wordpress-mobile/WordPress-iOS/pull/21654#discussion_r1340635422) localized text.

With the changes, I prioritize showing as much information as possible and avoiding cutting off any text:

- Changed the text back to "Free for the first year with annual paid plans"
- Moved this long text to the left side under the domain name, as is the case on Android (https://github.com/wordpress-mobile/WordPress-Android/pull/19283) and the Web
- Adjusted priorities, so the left side would expand as long as price fits, and then wrap to another line

| short domain - short price | long domain - long price | short domain - long price | short - dark mode | short - dynamic type |
| --- | --- | --- | --- | --- |
| ![short domain - short price](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/7d3e1ef9-a75d-436a-be9e-cca7fc906cb0) | ![long domain - long price](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/ef2d34e0-27ce-4ddf-b774-b46395acde11) | ![short domain - long price](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/27e50d3d-8bf9-4b5f-920c-070926eed04c) | ![short - dark mode](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/49377456-25d3-401c-a412-ceac06b0b8bb) | ![short - dynamic type](https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/5398355f-9f44-4292-b70c-f00b7095da4e) |

- Additionally, made VoiceOver improvements

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/41040a8c-8cd1-4ce5-8445-762202c8da92

- And added right-to-left language fixes

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/59a03f63-b3e2-4de8-9ed7-24488175f821" width="250" alt="image">

### To test:

1. Enable "Plans in Site creation feature flag"
2. Create new site (either with a fresh account, or via site selector)
3. Search for domains
4. Confirm that "Free for the first year with annual paid plans" label is shown

#### Regression

1. Disable "Plans in Site creation feature flag"
2. Create new site (either with a fresh account, or via site selector)
3. Search for domains
4. Confirm that free domains are continued to be shown in a one-line cell

## Regression Notes
1. Potential unintended areas of impact

Breaking already existing UI for the cell

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

staskus committed:
Show free for the first year with annual paid plans label under the domain

staskus committed:
Rename domain AddressTableViewCell labels

staskus committed:
VoiceOver read the whole cell at the same time

staskus committed:
Adjust placement of dot view for right to left languages

guarani commented on WordPress/Classes/ViewRelated/Site Creation/Web Address/AddressTableViewCell.swift:
```diff
@@ -160,10 +177,25 @@ final class AddressTableViewCell: UITableViewCell {
+        case .freeWithPaidPlan:
+            let firstYearAttributes: [NSAttributedString.Key: Any] = [
+                .foregroundColor: Appearance.saleCostTextColor,
+                .font: Appearance.smallCostFont
```
nitpick: I noticed the "Free for the first year..." text doesn't update with larger text sizes, but uses the `.footnote` size which _I think_ should still update, although I haven't checked this. 

![Screenshot 2023-10-04 at 15 57 49](https://github.com/wordpress-mobile/WordPress-iOS/assets/1898325/1a1f05af-318e-44ab-bb4c-b20c830895d5)

staskus replied:
Yes, you're right. We're using `NSAttributedString` with `UIFont` attributes. It requires the view to be reopened for font size changes to react or to listen to notifications and rebuild attributed string. However, some labels do not have a specific font set for attributed text, and have font only set on `UILabel`. This way the dynamic type changes work immediately. 

I made an update to rely on the font set directly on the label rather than on the attributed text where we can. I want to avoid listening to notifications in the `UITableViewCell`. Therefore, in the most common cases we now react to dynamic type changes immediately:

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/8f7a2697-30b0-4ba6-8559-03551985dd90

guarani reviewed (approved):
Working well, I see the updated copy 👍 
I tested with smaller/larger text sizes and have one observation (not a blocker). I didn't test longer prices or RTL text, but could do that if you feel it's worth testing.

guarani commented:
I tested RTL now as well and it's working 👍

staskus committed:
Merge branch 'trunk' into task/21635-plans-in-site-creation-update-domain-selection-cell-ui-for-long-text

staskus committed:
Better support for dynamic type and attributed text

If we set font on attributed text, we need to listen to notifications to automatically update dynamic type, otherwise the updated font will only appear after reopening the view.

To avoid listening to notifications in UITableViewCell, making most common attributed text cases rely on the font set directly on the label, and not on attributed text. This way dynamic type updates work out of the box.
